extensions/resource-deployment/src/test/resourceTypeService.test.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import 'mocha';
+import * as TypeMoq from 'typemoq';
+import assert = require('assert');
+import { EOL } from 'os';
+import { ResourceTypeService } from '../services/resourceTypeService';
+import { IPlatformService } from '../services/platformService';
+import { ToolsService } from '../services/toolsService';
+
+suite('Resource Type Service Tests', function (): void {
+
+	test('test resource types', () => {
+		const mockPlatformService = TypeMoq.Mock.ofType<IPlatformService>();
+		const toolsService = new ToolsService();
+		const resourceTypeService = new ResourceTypeService(mockPlatformService.object, toolsService);
+		// index 0: platform name, index 1: number of expected resource types
```

Comment:
>// index 0: platform name, index 1: number of expected resource types [](start = 2, length = 69)

FYI for this kind of thing I prefer to define an interface and use that instead. And it would seem to make more sense having it be a single array of those types :

interface PlatformInfo {
   name: string;
   numberExpectedResourceTypes: number;
}

...

const platforms: PlatformInfo[] = [ {name: 'win32', numberExpectedResourceTypes: 2}, ....]

It's just a cleaner way of exposing it IMO